src/app/modules/analytics-config/components/edit-histogram-label/edit-histogram-label.component.ts

```diff
@@ -0,0 +1,229 @@
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+*/
+import { Component, Input, OnDestroy, OnInit } from '@angular/core';
+import { AbstractControl, FormArray, FormControl, FormGroup } from '@angular/forms';
+import { marker } from '@biesbjerg/ngx-translate-extract-marker';
+import { BehaviorSubject, combineLatest, distinctUntilChanged, filter, Subject, takeUntil } from 'rxjs';
+import { MainFormService } from '@services/main-form/main-form.service';
+
+interface LabelConfig {
+    title: string;
+    dataFieldValue: AbstractControl;
+    dataFieldFunction?: AbstractControl;
+    labelControl: AbstractControl;
+    unitControl: AbstractControl;
+    message?: BehaviorSubject<string>;
+    readonly?: BehaviorSubject<boolean>;
+}
+
+@Component({
+  selector: 'arlas-edit-histogram-label',
+  templateUrl: './edit-histogram-label.component.html',
+  styleUrls: ['./edit-histogram-label.component.scss']
+})
+export class EditHistogramLabelComponent implements OnInit, OnDestroy {
+    /**
+     * field from parent form control
+     */
+    @Input() public dataStep: { aggregation: FormGroup; metric: FormGroup; collection: FormControl; };
+    /**
+     * field from parent form control
+     */
+    @Input() public unmanagedFieldRenderStep: {
+        chartXLabel: FormControl;
+        chartYLabel: FormControl;
+        xUnit: FormControl;
+        yUnit: FormControl;
+    };
+    @Input() public data: LabelConfig[];
+    public displayedColumns = ['recap', 'fieldLabel', 'fieldColumn', 'fieldUnits'];
+    /**
+     *  X axis config
+     * @protected
+     */
+    protected xLabelConfig: LabelConfig;
+    /**
+     *  Y axis config
+     * @protected
+     */
+    protected yLabelConfig: LabelConfig;
+    /**
+     * Get lookAndField value to listen collection unit change
+     * @protected
+     */
+    protected lookAndFeelFormControl: AbstractControl<any>;
+    /**
+     * Comparing value
+     * @protected
+     */
+    protected currentCollectionName: string;
+
+    private _destroyed$ = new Subject();
+
+    public constructor(private mainFromService: MainFormService) {
+    }
+
+    public ngOnInit(): void {
+      if (!this.dataStep && !this.unmanagedFieldRenderStep) {
+        console.error('no config found');
+        return;
+      }
+
+      this.currentCollectionName = this.dataStep.collection.value;
+
+      this.xLabelConfig = {
+        title: marker('x-label'),
+        dataFieldValue: this.dataStep.aggregation.get('aggregationField'),
+        labelControl: this.unmanagedFieldRenderStep.chartXLabel,
+        unitControl: this.unmanagedFieldRenderStep.xUnit,
+        message: new BehaviorSubject(''),
+        readonly: new BehaviorSubject(false)
+      };
+
+      this.yLabelConfig = {
+        title: marker('y-label'),
```

Comment:
I prefer naming it : y-axis

also, can you apply the init and extract of markers to add the translation of those texts ?